src/shogun/structure/GEMPLP.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2014 Jiaolong Xu
+ * Copyright (C) 2014 Jiaolong Xu
+ */
+
+#ifndef __GEMPLP_H__
+#define __GEMPLP_H__
+
+#include <shogun/lib/config.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/structure/FactorGraph.h>
+#include <shogun/structure/Factor.h>
+#include <shogun/structure/MAPInference.h>
+#include <shogun/lib/SGNDArray.h>
+
+#include <vector>
+#include <set>
```

Comment:
I understand, however, that we can use gemplp from Python since you have used it in your notebook, right? I guess because you an enumerate or similar mechanism to select the type of approximate inference to use in your MAP class.

That is perfect, then! The interface, in this case, MAPInference class, is accessible to swig and through it one can define whether to use gemplp, etc.